build: don't depend on npm packages from public api tests

IgorMinar opened the pull request:
we need only the inputs from ng_module.

the missing piece is to ensure that we depend on all secondary entry points

IgorMinar committed:
build: don't depend on npm packages from public api tests

We need only the inputs from ng_module.

IgorMinar commented:
cc: @petebacondarwin

IgorMinar committed:
build: fix the accept binary path in ts-api-guardian error message

Fixes #26589

IgorMinar committed:
build: remove unused @angular/compiler ts-api-guardian golden files

these files are not used because the api surface of this package is not considered
to be stable/public api.

the presence of these files only confuses the reader.

petebacondarwin commented on tools/ts-api-guardian/lib/cli.ts:
```diff
@@ -66,7 +66,7 @@ export function startCli() {
         if (!!process.env['BAZEL_TARGET']) {
           console.error('\n\nAccept the new golden file:');
-          console.error(`  bazel run ${process.env['BAZEL_TARGET']}.accept`);
+          console.error(`  bazel run ${process.env['BAZEL_TARGET'].replace(/_bin$/, "")}.accept`);
```
💯

petebacondarwin commented on tools/public_api_guard/public_api_guard.bzl:
```diff
@@ -0,0 +1,23 @@
+        [package_name, entry_point_tail] = entry_point.split("/", 1)
+        directory_name = entry_point_tail.split("/")[-1]
+        target_suffix = "/" + entry_point_tail if package_name != entry_point_tail else ""
+        actual_file = "packages/%s/%s/%s.d.ts" % (package_name, target_suffix, directory_name)
```
if `target_suffix` is `""` then won't we end up with `actual_file` looking like `packages/common//testing.d.ts` (note the double slash)?

petebacondarwin replied:
Actually if the `target_suffix` is not empty then it must start with a `/` right?

petebacondarwin replied:
So you'll end up with a double slash either way.

mhevery reviewed (approved):
LGTM, please answers comments from others.

IgorMinar committed:
test: mark public_api_guard test with fixme-ivy-aot and no-ivy-jit

These tests are currently broken because of the following reasons:
- ivy no longer emits a generated index, so the filename of the main d.ts file is different
- ivy currently exports some symbols that don't match the golden file, this needs investigation
